core/src/blocks/google_answer.rs

```diff
@@ -0,0 +1,126 @@
+use crate::blocks::block::{parse_pair, replace_variables_in_string, Block, BlockType, Env};
+use crate::Rule;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use pest::iterators::Pair;
+use serde_json::Value;
+use serde_json::json;
+use hyper::{body::Buf, Body, Client, Method, Request};
+use hyper_tls::HttpsConnector;
+use std::io::prelude::*;
+use urlencoding::encode;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone)]
+pub struct GoogleAnswer {
+    question: String,
+    // hash: String,
+}
+
+impl GoogleAnswer {
```

Comment:
What's the motivation to go just for the answer? I think we want the full experience no and get the raw SERPAPI reply?

Reply:
Yes what you propose seems ideal.

Proposal (cc @mauriyouth)
- I take over both your PR to merge them together this afternoon. Your SerpAPI integration, @mauriyouth work to link it to the frontend
- v0 is a raw JSON out of Serp
- From there we iterate to add options to that block to provide more direct use-cases with less code to write?

wdyt?

Reply:
good for me 👍

Reply:
I'm already working on a front-end of the Answers version, have it mostly working (still messing with styles, allowing direct enter of SerpApi key on the Dust app page, and a bug that is unearthed in the LLM block UI by adding a non-LLM provider). Do you want me to finish that up in the next couple hours before you take this over? 
![image](https://user-images.githubusercontent.com/44199/196981102-bded8d91-735c-4070-a09f-1a46618064a8.png)